src/widgets/dialogs/EditHotkeyDialog.cpp

```diff
@@ -0,0 +1,296 @@
+#include "widgets/dialogs/EditHotkeyDialog.hpp"
+
+#include "Application.hpp"
+#include "common/QLogging.hpp"
+#include "controllers/hotkeys/ActionNames.hpp"
+#include "controllers/hotkeys/HotkeyController.hpp"
+#include "controllers/hotkeys/HotkeyHelpers.hpp"
+#include "ui_EditHotkeyDialog.h"
+
+namespace chatterino {
+
+EditHotkeyDialog::EditHotkeyDialog(const std::shared_ptr<Hotkey> hotkey,
+                                   bool isAdd, QWidget *parent)
+    : QDialog(parent, Qt::WindowStaysOnTopHint)
+    , ui_(new Ui::EditHotkeyDialog)
+    , data_(hotkey)
+{
+    this->ui_->setupUi(this);
+    const auto app = getApp();
+    // dynamically add category names to the category picker
+    for (const auto &[_, hotkeyCategory] : app->hotkeys->categories())
```

Comment:
```suggestion
    for (const auto &[_, hotkeyCategory] : getApp()->hotkeys->categories())
```

Reply:
Should be resolved with the other comment here